flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/functions/hive/conversion/HiveInspectors.java

```diff
@@ -0,0 +1,248 @@
+				if (((JavaFloatObjectInspector) inspector).preferWritable()) {
+					return o -> new FloatWritable((Float) o);
+				} else {
+					return IdentityConversion.INSTANCE;
+				}
+			} else if (inspector instanceof JavaDoubleObjectInspector) {
+				if (((JavaDoubleObjectInspector) inspector).preferWritable()) {
+					return o -> new DoubleWritable((Double) o);
+				} else {
+					return IdentityConversion.INSTANCE;
+				}
+			}
+		}
+
+		throw new FlinkHiveUDFException(
+			String.format("Flink doesn't support convert object conversion for %s yet", inspector));
+	}
+
+	/**
+	 * Unwraps an object from Hive object to Flink object with an ObjectInspector.
+	 */
+	public static Object unwrap(ObjectInspector inspector, Object data) {
+		if (data == null) {
+			return null;
+		}
+
+		if (inspector instanceof VoidObjectInspector) {
+			return null;
+		}
+
+		if (inspector instanceof PrimitiveObjectInspector) {
+			if (inspector instanceof BooleanObjectInspector) {
+				BooleanObjectInspector oi = (BooleanObjectInspector) inspector;
+
+				return oi.preferWritable() ?
+					oi.get(data) :
+					oi.getPrimitiveJavaObject(data);
+			} else if (inspector instanceof StringObjectInspector) {
+				StringObjectInspector oi = (StringObjectInspector) inspector;
+
+				return oi.preferWritable() ?
+					oi.getPrimitiveWritableObject(data).toString() :
+					oi.getPrimitiveJavaObject(data);
+			} else if (inspector instanceof ByteObjectInspector) {
+				ByteObjectInspector oi = (ByteObjectInspector) inspector;
+
+				return oi.preferWritable() ?
+					oi.get(data) :
+					oi.getPrimitiveJavaObject(data);
+			} else if (inspector instanceof ShortObjectInspector) {
+				ShortObjectInspector oi = (ShortObjectInspector) inspector;
+
+				return oi.preferWritable() ?
+					oi.get(data) :
+					oi.getPrimitiveWritableObject(data);
+			} else if (inspector instanceof IntObjectInspector) {
+				IntObjectInspector oi = (IntObjectInspector) inspector;
+
+				return oi.preferWritable() ?
+					oi.get(data) :
+					oi.getPrimitiveWritableObject(data);
+			} else if (inspector instanceof LongObjectInspector) {
+				LongObjectInspector oi = (LongObjectInspector) inspector;
+
+				return oi.preferWritable() ?
+					oi.get(data) :
+					oi.getPrimitiveWritableObject(data);
+			} else if (inspector instanceof FloatObjectInspector) {
+				FloatObjectInspector oi = (FloatObjectInspector) inspector;
+
+				return oi.preferWritable() ?
+					oi.get(data) :
+					oi.getPrimitiveWritableObject(data);
+			} else if (inspector instanceof DoubleObjectInspector) {
+				DoubleObjectInspector oi = (DoubleObjectInspector) inspector;
+
+				return oi.preferWritable() ?
+					oi.get(data) :
+					oi.getPrimitiveWritableObject(data);
+			}
+		}
+
+		// TODO: handle complex types like struct, list, and map
+
+		throw new FlinkHiveUDFException(
+			String.format("Unwrap does not support ObjectInspector '%s' yet", inspector));
+	}
+
+	public static ObjectInspector getObjectInspector(Class clazz) {
```

Comment:
Is this method used anywhere?

Reply:
it's used in `HiveSimpleUDF#getHiveResultType()`

Reply:
ok